src/erlcloud_config.erl

```diff
@@ -48,20 +51,20 @@
 %% Library initialization.
 %%------------------------------------------------------------------------------
 
--spec new(string(), string()) -> aws_config().
+-spec new(string(), string()) -> config().
```

Comment:
Maybe this will even start shifting `:new` uses away from `aws_config()` which is the desired outcome.